spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py

```diff
@@ -0,0 +1,298 @@
+# Fixtures
+# =============================================================================
+@pytest.fixture
+def connection_dialog_factory(request):
+    """Set up kernel connection dialog."""
+    class DialogFactory(object):
+        def get(self):
+            dialog = KernelConnectionDialog()
+            request.addfinalizer(dialog.close)
+            return dialog
+
+    def teardown():
+        """Clear existing-kernel config and keyring passwords"""
+        CONF.remove_section("existing-kernel")
+
+        try:
+            import keyring
+            keyring.set_password("existing_kernel",
+                                 "ssh_key_passphrase", "")
+            keyring.set_password("existing_kernel",
+                                 "ssh_password", "")
+        except Exception:
+            pass
+
+    request.addfinalizer(teardown)
+    return DialogFactory()
+
+
+# =============================================================================
+# Tests
+# =============================================================================
+def test_kernel_connection_dialog_remembers_input_ssh_passphrase(
+    qtbot, connection_dialog_factory
+):
+    """Test to ensure that the dialog remembers user's kernel connection
+       settings and ssh key passphrase when the user checks the
+       save checkbox"""
+
+    dlg = connection_dialog_factory.get()
+
+    cf_path = "test_cf_path"
+    un = "test_username"
+    hn = "test_hostname"
+    pn = 123
+    kf = "test_kf"
+    kfp = "test_kfp"
+
+    # fill out cf path
+    dlg.cf.clear()
+    qtbot.keyClicks(dlg.cf, cf_path)
+    # check the remote kernel box
+    dlg.rm_group.setChecked(True)
+    # fill out hostname
+    dlg.hn.clear()
+    qtbot.keyClicks(dlg.hn, hn)
+    # fill out port
+    dlg.pn.clear()
+    qtbot.keyClicks(dlg.pn, str(pn))
+    # fill out username
+    dlg.un.clear()
+    qtbot.keyClicks(dlg.un, un)
+
+    # select ssh keyfile radio
+    dlg.kf_radio.setChecked(True)
+    assert dlg.kf.isEnabled()
+
+    # fill out ssh keyfile
+    dlg.kf.clear()
+    qtbot.keyClicks(dlg.kf, kf)
+
+    # fill out passphrase
+    dlg.kfp.clear()
+    qtbot.keyClicks(dlg.kfp, kfp)
+
+    # check save connection settings
+    dlg.save_layout.setChecked(True)
+
+    # Press ok and save connection settings
+    qtbot.mouseClick(dlg.accept_btns.button(QDialogButtonBox.Ok),
+                     Qt.LeftButton)
+
+    # create new dialog and check fields
+    new_dlg = connection_dialog_factory.get()
+    assert new_dlg.cf.text() == cf_path
+    assert new_dlg.rm_group.isChecked()
+    assert new_dlg.hn.text() == hn
+    assert new_dlg.un.text() == un
+    assert new_dlg.pn.text() == str(pn)
+    assert new_dlg.kf.text() == kf
+    if (not sys.platform.startswith('linux') or
+            not os.environ.get('CI') is not None):
+        assert new_dlg.kfp.text() == kfp
+
+
+def test_kernel_connection_dialog_doesnt_remembers_input_ssh_passphrase(
+    qtbot, connection_dialog_factory
```

Comment:
8 space indent, "remembers" -> "remember", reduce length as above, put close paren on this line